content/Namespaces/Namespaces.q.md

```diff
@@ -0,0 +1,2010 @@
+
+ - It imports the `Constants` class in the `Math` namespace.
+
+ - It removes the `Constants` class in the `Math` namespace.
+
+ - It displays the `Constants` class in the `Math` namespace.
+
+
+/// type=SS, answer=[3]
+
+Which statement best describes `Geometry::getCircleArea(10)` on line 6 of `Main.php`?
+
+ - It removes the argument `10` from the `getCircleArea()` method of the `Geometry` class.
+
+ - It removes the static method `getCircleArea()` with the parameter `10` of the `Geometry` class.
+
+ - It calls the static method `getCircleArea()` passing the argument `10` outside the `Geometry` class.
+
+ - It creates the static method `getCircleArea()` with the parameter `10` outside the `Geometry` class.
+
+ - It defines the static method `getCircleArea()` with the parameter `10` outside the `Geometry` class.
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[Math.php,Constants.php,Circle.php,Main.php]
+
+```php
+<?php
+namespace Math;
+
+class Geometry
+{
+    const PI = 3.14159;
+	
+    static function getCircleArea($radius)
+    {
+        return self::PI * $radius ** 2;
+    }
+}
+?>
+```
+
+```php
+<?php
+namespace Math;
+
+class Constants
+{
+    const PI = 3.14159;
+}
+?>
+```
+
+```php
+<?php
+namespace Math\Geometry;
+
+use Math\Constants;
+
+class Circle
+{
+    public $radius;
+	
+    public function __construct($radius)
+    {
+        $this -> radius = $radius;
+    }
+	
+    public function getCircumference()
+    {
+        return 2 * Constants::PI * $this -> radius;
+    }
+}
+?>
+```
+
+```php
+<?php
+require_once('./Math.php');
+
+use Math\Geometry;
+
+echo "The area of the circle is: " . Geometry::getCircleArea(10);
+?>
+```
+/// type=SS, answer=[3]
+
+Remove the statement `use Math\Geometry;` on line 4 of `Main.php`. Execute the program. What is its output?
+
+ - It prints `10`.
+
+ - It prints `314`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `The area of the circle is: 314`.
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[Math.php,Constants.php,Circle.php,Main.php]
+
+```php
+<?php
+namespace Math;
+
+class Geometry
+{
+    const PI = 3.14159;
+	
+    static function getCircleArea($radius)
+    {
+        return self::PI * $radius ** 2;
+    }
+}
+?>
+```
+
+```php
+<?php
+namespace Math;
+
+class Constants
+{
+    const PI = 3.14159;
+}
+?>
+```
+
+```php
+<?php
+namespace Math\Geometry;
+
+use Math\Constants;
+
+class Circle
+{
+    public $radius;
+	
+    public function __construct($radius)
+    {
+        $this -> radius = $radius;
+    }
+	
+    public function getCircumference()
+    {
+        return 2 * Constants::PI * $this -> radius;
+    }
+}
+?>
+```
+
+```php
+<?php
+require_once('./Math.php');
+
+echo "The area of the circle is: " . Geometry::getCircleArea(10);
+?>
+```
+/// type=SS, answer=[5]
+
+On line 4 of `Main.php`, replace `Geometry::getCircleArea(10)` with `Math\Geometry::getCircleArea(10)`. Execute the program. What is its output?
```

Comment:
And this part also... I think this answers to __`how do you access a class globally without "use"? \Foo\Bar`__. Or do you need to have a statement that explains this in the facts? Please let me know your thoughts on this. Thank you.

Reply:
this is fine when accessing those classes from the default namespace but not when accessing from a different namespace and `use` isn't used

Reply:
I see, this is clear now to me. I will edit the content to add these information including the exceptions. Thank you.